File: 0283-move-zeroes/0283-move-zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        pos = 0
        for i in range(len(nums)):
            if nums[i] != 0:
                nums[pos], nums[i] = nums[i], nums[pos]
                pos += 1
        
        
        
        
        l = 0 
        while l < len(nums) and nums[l] != 0:
            l += 1
        if l == len(nums):
            return 
        r = l + 1
        
        while r < len(nums):
            if nums[r]:
                nums[l], nums[r] = nums[r], nums[l]
                l += 1
                r += 1
            while r < len(nums) and nums[r] == 0:
                r += 1
    
                
                
        
        

File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int pos = 0;
        int n = nums.length;
        
        for (int i = 0; i < n; i++){
            if (nums[i] != 0){
                int temp = nums[pos];
                nums[pos] = nums[i];
                nums[i] = temp;
                pos++;
            }
        }
    }
}